OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -48,10 +48,16 @@ public Player Owner
 		{
 			get
 			{
-				if (disguise.Disguised)
-					return self.Owner == self.World.LocalPlayer ? self.Owner : disguise.AsPlayer;
+				if (!disguise.Disguised)
+					return self.Owner;
 
-				return self.Owner;
+				Player p = null;
+				if (self.World.LocalPlayer != null && self.World.LocalPlayer.WinState == WinState.Undefined)
```

Comment:
I'd argue that any player with the disable shroud cheat should see thru the disguise since that cheat enables them already too see things like ore load in a harvester and refinery, as well as ammo loads etc.

And are you sure about `Everyone` being an ally to all players? If I remember correctly, it is neutral and that makes that second check necessary.